backend/layers/processing/process_validate.py

```diff
@@ -107,6 +108,23 @@ def populate_dataset_citation(
         adata.uns["citation"] = citation
         adata.write(adata_path, compression="gzip")
 
+    def get_spatial_metadata(self, spatial_dict: Dict[str, Any]) -> Optional[SpatialMetadata]:
+        """
+        Extracts spatial dataset metadata from the uns dict of an AnnData object
+
+        :param spatial_dict: the value of the 'spatial' key from the uns dict of an AnnData object
+        :return: SpatialMetadata object
+        """
+        is_single = spatial_dict.get("is_single")
+        has_fullres = False
+        # schema validation ensures nested 'fullres' key is only included when is_single is True
+        if is_single:
+            # schema validation ensures there can only be one other key in uns["spatial"] if "is_single" is True
+            library_id = [key for key in spatial_dict if key != "is_single"][0]
```

Comment:
This seems like a check that should be done by the validator and surface to it's consumers.  I know it's more work, but it is a good idea to avoid duplicating validation logic from one repo to another.

Reply:
True, this isn't really a check. My concern was introducing check logic into the ingest pipeline. Carry on.